### ISSUE TITLE
[Error] `host-spawn` is not installed automatically when creating a new distrobox

### ISSUE THREAD

**Issue by comminutus** [89luca89/distrobox#945]
**Is your feature request related to a problem? Please describe.**
I build my own OCI images for use with `distrobox`.  When I create a new distrobox from my custom image, any time I use `distrobox-host-exec` I'm prompted to download and install `host-spawn`.  I'd rather have `host-spawn` be part of my images.  

**Describe the solution you'd like**
Perhaps a command switch in `distrobox-create`, like `distrobox create -i my-image -n my-image --host-spawn` to automatically install it when the image launches.  Even better, package it into a package manager so that a Fedora image can install it with `dnf install host-spawn` - this way it will become part of the OCI image.

**Describe alternatives you've considered**
For now I'm lifting the download code from https://github.com/89luca89/distrobox/blob/main/distrobox-host-exec and putting this into my `Containerfile`:
```
RUN \
    export host_spawn_version=$(curl -sL "https://api.github.com/repos/1player/host-spawn/releases/latest" | jq -r ".tag_name")         && \
    curl -fsLSo /tmp/host-spawn "https://github.com/1player/host-spawn/releases/download/${host_spawn_version}/host-spawn-$(uname -m)"  && \
	mv /tmp/host-spawn /usr/bin/                                                                                                        && \
	chmod +x /usr/bin/host-spawn
```

**Comment by comminutus** [comment]
Strange, does `init` get called when using `distrobox create`? I'm getting the warning, "Warning: host-spawn not found or version is too old! Do you want to install host-spawn utility?" after using `distrobox create ...`, `distrobox enter ...` and then running a command that is sym-linked to `distrobox-host-exec`.  Or do I have to use the `--init` switch when creating the distrobox?

**Comment by comminutus** [comment]
@89luca89 I don't think that commit is doing what was intended. Take a look at this:
```
❯  distrobox --version && distrobox create && distrobox enter
distrobox: 1.5.0.2
Creating 'my-distrobox' using image registry.fedoraproject.org/fedora-toolbox:38	 [ OK ]
Distrobox 'my-distrobox' successfully created.
To enter, run:

distrobox enter my-distrobox

Container my-distrobox is not running.
Starting container my-distrobox
run this command to follow along:

 podman logs -f my-distrobox

 Starting container...                  	 [ OK ]
 Installing basic packages...           	 [ OK ]
 Setting up read-only mounts...         	 [ OK ]
 Setting up read-write mounts...        	 [ OK ]
 Setting up host's sockets integration...	 [ OK ]
 Integrating host's themes, icons, fonts...	 [ OK ]
 Setting up package manager exceptions...	 [ OK ]
 Setting up rpm exceptions...           	 [ OK ]
 Setting up sudo...                     	 [ OK ]
 Setting up groups...                   	 [ OK ]
 Setting up users...                    	 [ OK ]
 Integrating host's themes, icons, fonts...	 [ OK ]
 Setting up package manager exceptions...	 [ OK ]
 Setting up rpm exceptions...           	 [ OK ]
 Setting up sudo...                     	 [ OK ]
 Setting up groups...                   	 [ OK ]
 Setting up users...                    	 [ OK ]
 Executing init hooks...                	 [ OK ]

Container Setup Complete!
[me@my-distrobox dev]$ /usr/bin/distrobox-host-exec echo 'test'
Warning: host-spawn not found or version is too old!
Do you want to install host-spawn utility? [Y/n] 
```

**Comment by 89luca89** [comment]
![image](https://github.com/89luca89/distrobox/assets/598882/3632ba95-f533-4b85-ae30-a813131c1725)

Works for me...
Not sure why not for you, but definitely the command is there to download

**Comment by comminutus** [comment]
I debugged it a bit and it looks like the command is failing silently because errors are redirected to `/dev/null`. When I look at the error output, it says that you have to be within a container to run `distrobox-host-exec`. It says this because `CONTAINER_ID` is not defined. Upon searching the code for where `CONTAINER_ID` is set, it looks like it's only set inside `distrobox-enter`. If `distrobox-init` is invoked before `distrobox-enter` then it will fail. Since `distrobox-init` is called when `distrobox-create` is called, it will always fail. You must have `CONTAINER_ID` set?

**Comment by comminutus** [comment]
I debugged it a bit and it looks like the command is failing silently because errors are redirected to `/dev/null`: https://github.com/89luca89/distrobox/blob/0eeb4e97e07b1b37c6d50ab5f510966f90e2a743/distrobox-init#L890

I changed it locally to `/usr/bin/distrobox-host-exec -Y test` and then saw this error:
```
You must run  distrobox-host-exec inside a container!
```

This comes from here: https://github.com/89luca89/distrobox/blob/c0e0dceb40decc41511cb95bf2385139d5af5c18/distrobox-host-exec#L121 , which only occurs if `CONTAINER_ID` is not set.

Upon searching the code for where `CONTAINER_ID` is set, it looks like the only place where it's set is inside `distrobox-enter` around here: https://github.com/89luca89/distrobox/blob/0eeb4e97e07b1b37c6d50ab5f510966f90e2a743/distrobox-enter#L310 

Therefore, if `distrobox-init` is invoked before `distrobox-enter` then installing `host-spawn` will fail. Because `distrobox-init` is called only when `distrobox-create` is called, installing `host-spawn` will always fail unless the container has the `CONTAINER_ID` environment variable set.

To test this out, I ran `distrobox create --additional-flags "--env CONTAINER_ID=my-distrobox" && distrobox enter`. I noticed the `create` took much longer this time. Upon entering the shell, I ran `distrobox-host-exec test` and it didn't prompt me to install `host-spawn`. I also confirmed it was installed with:
```
which host-spawn
/usr/bin/host-spawn
```

Perhaps your installation always sets `CONTAINER_ID` prior to running `distrobox-init`?  I'd be happy to submit a PR, but I'm not sure what the right course of action is here.  Should the check for `CONTAINER_ID` just be removed?  If not, how should `CONTAINER_ID` be set within `distrobox-init`?
